test/restapitest/webhook_test.go

```diff
@@ -0,0 +1,223 @@
+	}{
+		"without OnRemoteWebhook, default webhook path": {
+			reqPath:    "/webhook",
+			listenPath: "/webhook",
+			called:     true,
+		},
+		"without OnRemoteWebhook, long request path": {
+			reqPath:    "/webhook/request-suffix",
+			listenPath: "/webhook/request-suffix",
+			called:     true,
+		},
+		"with OnRemoteWebhook, default request path": {
+			onRemoteWebhookPath: "/my-webhook",
+			reqPath:             "/webhook",
+			listenPath:          "/my-webhook",
+			called:              true,
+		},
+		"with OnRemoteWebhook, long request path": {
+			onRemoteWebhookPath: "/my-webhook",
+			reqPath:             "/webhook/request-suffix",
+			listenPath:          "/my-webhook/request-suffix",
+			called:              true,
+		},
+		"with OnRemoteWebhook long path": {
+			onRemoteWebhookPath: "/my-webhook/manifest-suffix",
+			reqPath:             "/webhook",
+			listenPath:          "/my-webhook/manifest-suffix",
+			called:              true,
+		},
+		"with OnRemoteWebhook long path, long request path": {
+			onRemoteWebhookPath: "/my-webhook/manifest-suffix",
+			reqPath:             "/webhook/request-suffix",
+			listenPath:          "/my-webhook/manifest-suffix/request-suffix",
+			called:              true,
+		},
+		"with OnRemoteWebhook, wrong request path": {
+			onRemoteWebhookPath: "/my-webhook",
+			reqPath:             "/my-webhook",
+			listenPath:          "/my-webhook",
+			called:              false,
+		},
+		"with OnRemoteWebhook, long wrong request path": {
+			onRemoteWebhookPath: "/my-webhook",
+			reqPath:             "/my-webhook/request-suffix",
+			listenPath:          "/my-webhook/request-suffix",
+			called:              false,
+		},
+		"with OnRemoteWebhook long path, wrong request path": {
+			onRemoteWebhookPath: "/my-webhook/manifest-suffix",
+			reqPath:             "/my-webhook/manifest-suffix",
+			listenPath:          "/my-webhook/manifest-suffix",
+			called:              false,
+		},
+		"with OnRemoteWebhook long path, long wrong request path": {
+			onRemoteWebhookPath: "/my-webhook/manifest-suffix",
+			reqPath:             "/my-webhook/request-suffix",
+			listenPath:          "/my-webhook/manifest-suffix/request-suffix",
+			called:              false,
+		},
+	} {
+		th.Run(name, func(th *Helper) {
+			var onRemoteWebhook *apps.Call
+			if tc.onRemoteWebhookPath != "" {
+				onRemoteWebhook = apps.NewCall(tc.onRemoteWebhookPath)
+			}
+
+			app := newWebhookApp(th.T, onRemoteWebhook)
+			th.InstallAppWithCleanup(app)
+
+			calledChan := make(chan bool)
+
+			app.HandleCall(tc.listenPath, func(cr goapp.CallRequest) apps.CallResponse {
+				calledChan <- true
+				return apps.NewTextResponse("success")
+			})
+
+			webhookSecret := getTestWebhookSecret(th, app)
+			err := sendWebhookPostRequest(th, webhookSecret, tc.reqPath)
+			if err != nil && (tc.called || err.Error() != "404 page not found") {
+				th.Fatal("received error for webhook request: " + err.Error())
+			}
+
+			var result bool
```

Comment:
Is this consistent with how we check for notifications? I'd like to make it a bit more robust, but as a separate PR.

Reply:
@levb I'm definitely open for improvement here. The difference between this and notifications is that this isn't always called, so we can't just block and wait for the receive from the channel. Not saying you're saying otherwise, just pointing it out